packages/edit-widgets/src/index.js

```diff
@@ -44,15 +46,27 @@ export function initialize( id, settings ) {
  *
  * @param {string} id       ID of the root element to render the section in.
  * @param {Object} settings Block editor settings.
+ * @param {string} widgetID ID of the widget being rendered.
  */
-export function customizerInitialize( id, settings ) {
-	registerCoreBlocks();
-	if ( process.env.GUTENBERG_PHASE === 2 ) {
-		__experimentalRegisterExperimentalCoreBlocks( settings );
+export function customizerInitialize( id, settings, widgetID ) {
+	// The customizer can has many widgets, it should only register blocks once.
+	if ( ! registered ) {
+		registerCoreBlocks();
+		if ( process.env.GUTENBERG_PHASE === 2 ) {
+			__experimentalRegisterExperimentalCoreBlocks( settings );
+		}
```

Comment:
We'll also need `registerBlock( legacyWidget );` as in line 33 in the same file. Maybe it's time to that logic into our own `registerWidgetsEditorBlocks()` function?